envoy/router/cluster_specifier_plugin.h

```diff
@@ -22,11 +22,11 @@ class ClusterSpecifierPlugin {
   /**
    * Create route from related route entry and request headers.
    *
-   * @param parent related route entry.
+   * @param parent related route.
    * @param header request headers.
    * @return RouteConstSharedPtr final route with specific cluster.
    */
-  virtual RouteConstSharedPtr route(const RouteEntry& parent,
+  virtual RouteConstSharedPtr route(RouteConstSharedPtr parent,
```

Comment:
comment: by this way, we the cluster specifier plugin could take the ownership of the parent route.